### PR TITLE
Avoid Reexporter when loader supports index fallback.

### PR DESCRIPTION
**Pull request by rwjblue**

### CODE REVIEW

**Comment by rwjblue** [comment]
@stefanpenner - Are you generally :+1: on the goal of this?

**Comment by stefanpenner** [comment]
> Are you generally  on the goal of this?

:+1:

File: lib/models/addon.js [hunk 1]
```diff
@@ -585,9 +585,15 @@ Addon.prototype.compileTemplates = function(tree) {
...
   this._requireBuildPackages();
 
+  var reexported;
+
   var addonJs = this.processedAddonJsFiles(tree);
   var templatesTree = this.compileTemplates(tree);
-  var reexported = new Reexporter(this.name, this.name + '.js');
+
+  if (!this.project._loaderSupportsIndexFallback()) {
+    reexported = new Reexporter(this.name, this.name + '.js');
```
Comment (stefanpenner) [inline lib/models/addon.js]: we may want to warn, encouraging people to upgrade.

Comment (rwjblue) [inline lib/models/addon.js]: Agreed, just added the warning.

**Comment by rwjblue** [comment]
Just did some smoke testing both with and without loader at 3.5.0 and all seems to work well.

**Comment by rwjblue** [comment]
OK, I believe that this is ready for review.  There is an iojs failure in Travis (and a random Node 5 failure in Appveyor), but those seem unrelated (also affecting https://github.com/ember-cli/ember-cli/pull/5172).

**Comment by stefanpenner** [comment]
I'll try and take a look at the general failures this morning. You can merge now, or wait till we are green again

**Comment by stefanpenner** [comment]
I'll try and take a look at the general failures this morning. You can merge now, or wait till we are green again

**Comment by rwjblue** [comment]
Being non-green makes me nervous (though I feel this PR is solid), I'd rather hold off until we have at least figured out why the build is failing (I haven't tested locally in iojs yet to see if it replicates the there).

**Comment by stefanpenner** [comment]
> I'd rather hold off until we have at least figured out why the build is failing (I haven't tested locally in iojs yet to see if it replicates the there).

:+1:

**Comment by rwjblue** [comment]
@homu r+

**Pull request merged**